### ISSUE TITLE
Batched gather (take_along_axis) produces incorrect results

### ISSUE THREAD

**Issue by tillahoffmann** [tillahoffmann/jax-mps#60]
## Description

`jnp.take_along_axis` with batched indices produces incorrect results on MPS. The gather operation uses the wrong batch index.

## Reproduction

```python
import jax
import jax.numpy as jnp
from jax import random

key = random.key(0)
probs = random.dirichlet(key, jnp.ones(5), (3,))
indices = jnp.array([2, 3, 3])

# Expected: gather probs[i, indices[i]] for each i
# probs[0, 2], probs[1, 3], probs[2, 3]

with jax.default_device(jax.devices('cpu')[0]):
    cpu = jnp.take_along_axis(probs, indices[:, None], axis=1)[:, 0]
    print(f'CPU: {cpu}')  # [0.061944, 0.36462006, 0.6308125]

with jax.default_device(jax.devices('mps')[0]):
    mps = jnp.take_along_axis(probs, indices[:, None], axis=1)[:, 0]
    print(f'MPS: {mps}')  # [0.061944, 0.42240828, 0.42240828] - WRONG!
```

## Impact

- Affects numpyro `CategoricalProbs` and `MultinomialProbs` distributions
- The `log_prob` computation returns incorrect values for batched inputs
- Currently masked by xfails in tests

## Analysis

MPS appears to use batch index 0 for all rows instead of the correct batch indices [0, 1, 2].
